internal/pkg/auth/user_login.go

```diff
@@ -60,22 +65,31 @@ func AuthorizeUser(p *print.Printer, isReauthentication bool) error {
 		}
 	}
 
-	listener, err := net.Listen("tcp", ":0")
-	if err != nil {
-		return fmt.Errorf("bind port for login redirect: %w", err)
+	var redirectURL string
+	var listener net.Listener
+	var listenerErr error
+	var port int
+	for i := range configuredPortRange {
+		port = defaultPort + i
+		portString := fmt.Sprintf(":%s", strconv.Itoa(port))
+		listener, listenerErr = net.Listen("tcp", portString)
+		if listenerErr == nil {
+			redirectURL = fmt.Sprintf("http://localhost:%d", port)
+			p.Debug(print.DebugLevel, "bound port %d for login redirect", port)
+			break
+		}
+		p.Debug(print.DebugLevel, "unable to bind port %d for login redirect: %s", port, listenerErr)
 	}
-	address, ok := listener.Addr().(*net.TCPAddr)
-	if !ok {
-		return fmt.Errorf("assert listener address type to TCP address")
+	if listenerErr != nil {
+		return fmt.Errorf("unable to bind port for login redirect, tried from port %d to %d: %w", defaultPort, port, err)
 	}
-	redirectURL := fmt.Sprintf("http://localhost:%d", address.Port)
 
 	conf := &oauth2.Config{
 		ClientID: cliClientID,
 		Endpoint: oauth2.Endpoint{
 			AuthURL: fmt.Sprintf("%s/authorize", idpEndpoint),
 		},
-		Scopes:      []string{"openid"},
+		Scopes:      []string{"openid offline_access"},
```

Comment:
What is this?

Reply:
The `offline_access` scope signals that we are also requesting a refresh token, that can be used to refresh the access token (https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess)